### PR TITLE
Bump to golang 1.18

### PR DESCRIPTION
**Pull request by SchSeba**

### CODE REVIEW

**Comment by zeeke** [comment]
LGTM

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -9,10 +9,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
 
-    - name: Set up Go 1.17
+    - name: Set up Go 1.18
       uses: actions/setup-go@v2
       with:
-        go-version: 1.17.8
+        go-version: 1.18.2
```
Comment (e0ne) [inline .github/workflows/go.yml:15]: Do we want to specify exact version here and just 1.18 in Dockerfile? IMO, it's enough to set 1.18 everywhere

**Review by e0ne — CHANGES_REQUESTED** [review]

File: Makefile [hunk 2]
```diff
@@ -44,7 +44,7 @@ BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
 # Image URL to use all building/pushing image targets
 IMG ?= controller:latest
 # Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
-CRD_OPTIONS ?= "crd:crdVersions={v1},trivialVersions=true"
+CRD_OPTIONS ?= "crd:crdVersions={v1}"
```
Comment (adrianchiris) [inline Makefile:47]: This change was due to controller-gen version bump ?

File: config/crd/bases/sriovnetwork.openshift.io_sriovibnetworks.yaml [hunk 3]
```diff
@@ -1,10 +1,9 @@
-
 ---
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.4.1
+    controller-gen.kubebuilder.io/version: v0.9.0
```
Comment (adrianchiris) [inline config/crd/bases/sriovnetwork.openshift.io_sriovibnetworks.yaml:6]: why was this change introduced ? (when bumping go 1.18 needed to bump contoller-gen kubebuilder ?)

File: Makefile (hunk 2 above)
Comment (SchSeba) [reply Makefile:47]: right

File: config/crd/bases/sriovnetwork.openshift.io_sriovibnetworks.yaml (hunk 3 above)
Comment (SchSeba) [reply config/crd/bases/sriovnetwork.openshift.io_sriovibnetworks.yaml:6]: yep I was not able to make the controller-gen version 0.4.1 to work with golang 1.18 so I had to bump it

File: .github/workflows/go.yml (hunk 1 above)
Comment (SchSeba) [reply .github/workflows/go.yml]: change it to v3 and use 1.18.2 I was like to be specific that it must be >= 1.18.2

**Comment by SchSeba** [comment]
fixing also the kustomize using the newer version

before the change

```
make deploy-setup-k8s                       
if ! which skopeo; then if [ -z  ]; then if [ -f /etc/redhat-release ]; then dnf -y install skopeo; elif [ -f /etc/lsb-release ]; then sudo apt-get -y update; sudo apt-get -y install skopeo; fi; fi; fi
/usr/bin/skopeo
/home/sscheink/Documents/GolangProjects/src/github.com/k8snetworkplumbingwg/sriov-network-operator/bin/controller-gen "crd:crdVersions={v1}" webhook paths="./..." output:crd:artifacts:config=./config/crd/bases
go: creating new go.mod: module tmp
Downloading sigs.k8s.io/kustomize/kustomize/v3@v3.8.7
go: downloading sigs.k8s.io/kustomize/kustomize/v3 v3.8.7
go: sigs.k8s.io/kustomize/kustomize/v3@v3.8.7 (in sigs.k8s.io/kustomize/kustomize/v3@v3.8.7):
	The go.mod file for the module providing named packages contains one or
	more exclude directives. It must not contain directives that would cause
	it to be interpreted differently than if it were the main module.
make: *** [Makefile:152: kustomize] Error 1
```

after the change

```
make deploy-setup-k8s
if ! which skopeo; then if [ -z  ]; then if [ -f /etc/redhat-release ]; then dnf -y install skopeo; elif [ -f /etc/lsb-release ]; then sudo apt-get -y update; sudo apt-get -y install skopeo; fi; fi; fi
/usr/bin/skopeo
/home/sscheink/Documents/GolangProjects/src/github.com/k8snetworkplumbingwg/sriov-network-operator/bin/controller-gen "crd:crdVersions={v1}" webhook paths="./..." output:crd:artifacts:config=./config/crd/bases
go: creating new go.mod: module tmp
Downloading sigs.k8s.io/kustomize/kustomize/v4@v4.5.5
go: downloading sigs.k8s.io/kustomize/kustomize/v4 v4.5.5
go: downloading github.com/spf13/cobra v1.4.0
go: downloading sigs.k8s.io/kustomize/cmd/config v0.10.7
go: downloading sigs.k8s.io/kustomize/api v0.11.5
go: downloading sigs.k8s.io/kustomize/kyaml v0.13.7
go: downloading github.com/spf13/pflag v1.0.5
go: downloading sigs.k8s.io/yaml v1.2.0
go: downloading golang.org/x/text v0.3.7
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/go-errors/errors v1.0.1
go: downloading github.com/olekukonko/tablewriter v0.0.4
go: downloading k8s.io/kube-openapi v0.0.0-20220401212409-b28bf2818661
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/monochromegane/go-gitignore v0.0.0-20200626010858-205db1a8cc00
go: downloading github.com/xlab/treeprint v0.0.0-20181112141820-a009c3971eca
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/google/gnostic v0.5.7-v3refs
go: downloading google.golang.org/protobuf v1.28.0
go: downloading github.com/evanphx/json-patch v4.11.0+incompatible
go: downloading github.com/imdario/mergo v0.3.5
go: downloading gopkg.in/inf.v0 v0.9.1
go: downloading github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510
go: downloading github.com/mattn/go-runewidth v0.0.7
go: downloading go.starlark.net v0.0.0-20200306205701-8dd3e2ee1dd5
go: downloading gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
go: downloading github.com/asaskevich/govalidator v0.0.0-20190424111038-f61b66f89f4a
go: downloading github.com/mitchellh/mapstructure v1.4.1
go: downloading k8s.io/utils v0.0.0-20210802155522-efc7438f0176
go: downloading github.com/golang/protobuf v1.5.2
go: downloading github.com/go-openapi/jsonreference v0.19.3
go: downloading github.com/go-openapi/swag v0.19.5
go: downloading github.com/PuerkitoBio/purell v1.1.1
go: downloading github.com/go-openapi/jsonpointer v0.19.5
go: downloading github.com/mailru/easyjson v0.7.0
go: downloading github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578
go: downloading golang.org/x/net v0.0.0-20210405180319-a5a99cb37ef4
/home/sscheink/Documents/GolangProjects/src/github.com/k8snetworkplumbingwg/sriov-network-operator/bin/kustomize build config/crd | kubectl apply -f -
customresourcedefinition.apiextensions.k8s.io/sriovibnetworks.sriovnetwork.openshift.io created
customresourcedefinition.apiextensions.k8s.io/sriovnetworknodepolicies.sriovnetwork.openshift.io created
customresourcedefinition.apiextensions.k8s.io/sriovnetworknodestates.sriovnetwork.openshift.io created
customresourcedefinition.apiextensions.k8s.io/sriovnetworkpoolconfigs.sriovnetwork.openshift.io created
customresourcedefinition.apiextensions.k8s.io/sriovnetworks.sriovnetwork.openshift.io created
customresourcedefinition.apiextensions.k8s.io/sriovoperatorconfigs.sriovnetwork.openshift.io created
hack/deploy-setup.sh sriov-network-operator
~/Documents/GolangProjects/src/github.com/k8snetworkplumbingwg/sriov-network-operator/deploy ~/Documents/GolangProjects/src/github.com/k8snetworkplumbingwg/sriov-network-operator
serviceaccount/sriov-network-operator created
serviceaccount/sriov-network-config-daemon created
role.rbac.authorization.k8s.io/sriov-network-operator created
role.rbac.authorization.k8s.io/sriov-network-config-daemon created
role.rbac.authorization.k8s.io/operator-webhook-sa created
rolebinding.rbac.authorization.k8s.io/sriov-network-operator created
rolebinding.rbac.authorization.k8s.io/sriov-network-config-daemon created
rolebinding.rbac.authorization.k8s.io/operator-webhook-sa created
clusterrole.rbac.authorization.k8s.io/sriov-network-operator unchanged
clusterrole.rbac.authorization.k8s.io/sriov-network-config-daemon unchanged
clusterrolebinding.rbac.authorization.k8s.io/sriov-network-operator configured
clusterrolebinding.rbac.authorization.k8s.io/sriov-network-config-daemon configured
configmap/supported-nic-ids created
Warning: would violate PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "sriov-network-operator" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "sriov-network-operator" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "sriov-network-operator" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "sriov-network-operator" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
deployment.apps/sriov-network-operator created
~/Documents/GolangProjects/src/github.com/k8snetworkplumbingwg/sriov-network-operator
```

**Review by adrianchiris — APPROVED** [review]

**Pull request merged**